offerzen/global/company-fast-track-access/company-prospect-form@v0.js

```diff
@@ -0,0 +1,229 @@
+(function () {
+  // FORM TARGETTING
+  const prospectFormId = '#Company-Prospect-Form';
+  const pageRef = 'onSubmitCompanyProspectForm';
+  const formSubmitButtonSelector = '#submit-button';
+  const formContainerClass = '.js-company-prospect-form';
+  // END FORM TARGETING
+  // This needs to run regardless of whether JQuery has loaded
+  // Disable button
+  const formSubmitButton = document.querySelector(
+    `${prospectFormId} ${formSubmitButtonSelector}`
+  );
+
+  // Store button labels for state changes
+  let initialButtonValue = formSubmitButton.getAttribute('data-initial');
+  let dataWait = formSubmitButton.getAttribute('data-wait');
+  let dataBusy = formSubmitButton.getAttribute('data-busy');
+
+  function disableSubmitButton(buttonLabel) {
+    formSubmitButton.setAttribute('disabled', 'disabled');
+    formSubmitButton.value = buttonLabel;
+  }
+
+  function enableSubmitButton() {
+    formSubmitButton.removeAttribute('disabled');
+    formSubmitButton.value = initialButtonValue;
+  }
+
+  disableSubmitButton('Loading...');
+  // ------------------------------------------------------------------------------------------
+  window.$loaded(function () {
+    window.$parsleyLoaded = function (cb) {
+      setTimeout(() => {
+        if (window.Parsley && window.ParsleyValidator) {
+          cb(window, document, parsley, undefined);
+          return;
+        }
+        $parsleyLoaded(cb);
+      }, 50);
+    };
+
+    window.$parsleyLoaded(function (window, document, parsley) {
+      const form = $(prospectFormId);
+      const formContainer = $(formContainerClass);
+
+      function trackSubmission() {
+        var emailValue = form.find('#email').val();
+        var isPlaywrightTest =
+          emailValue.match(/^\s*playwrighttest@offerzen\.com\s*$/i) != null;
+        if (!isPlaywrightTest) {
+          var event = form.find('.js-analytics-event').text();
+          var action = form.find('.js-analytics-action').text();
+          var label = form.find('.js-analytics-label').text();
+          var category = form.find('.js-analytics-category').text();
+          var source = form.find('.js-analytics-source').text();
+          dataLayer.push({
+            event: event || 'Company Lead Form Submitted',
+            action: action || 'Lead Form Submitted',
+            label: label || 'Company Sign Up / Employer Landing Page',
+            category: category || 'Core',
+            source: source || 'Demand Sign Up',
+          });
+        }
+      }
+
+      function startProspectPolling(prospectId, buttonLabelTimer) {
+        const errorText = formContainer.find('.js-form-error');
+
+        function poll() {
+          $.ajax({
+            type: 'GET',
+            url: `/api/company/prospects/${prospectId}`,
+            contentType: 'application/json',
+            headers: {
+              Accept: 'application/json',
+            },
+            statusCode: {
+              304: function () {
```

Comment:
Polling continues while server returns 304

Reply:
@dominicaw @jethroflanagan just sense check me over here, but shouldn't we clear the original timeout before we call it again on line 79?
Won't it be executing multiple timeouts if it's not cleared over here?

Reply:
@MarkusKuhn setTimeout is only called on `304` response and at the start of the function (last line). I don't think it's elsewhere. So it should only ever have one instance of the timeout in existence. Maybe I've missed something?